ci/tests/puppeteer/run.sh

```diff
@@ -220,13 +252,46 @@ if [[ "${REBUILD}" == "true" && "${RERUN}" != "true" ]]; then
   FLAGS=$(echo $BUILDFLAGS | sed 's/ //')
   printgreen "\nBuilding CAS found in $PWD for dependencies [${dependencies}] with flags [${FLAGS}]"
 
-  ./gradlew :webapp:cas-server-webapp-${project}:build \
-    -DskipNestedConfigMetadataGen=true -x check -x javadoc --build-cache --configure-on-demand --parallel \
-    ${BUILD_SCRIPT} ${DAEMON} -DcasModules="${dependencies}" -q ${BUILDFLAGS}
+  BUILD_CACHE=""
+  if [[ "$GRADLE_BUILDCACHE_PSW" != "" || "$CI" != "true" ]]; then
+    BUILD_CACHE="--build-cache"
+  fi
 
-  if [ $? -eq 1 ]; then
+  printcyan "Launching build in background to make observing slow builds easier"
+  targetArtifact=./webapp/cas-server-webapp-${project}/build/libs/cas-server-webapp-${project}-${casVersion}.${projectType}
+  if [[ -d ./webapp/cas-server-webapp-${project}/build/libs ]]; then
+    rm -rf ./webapp/cas-server-webapp-${project}/build/libs
+  fi
+  buildcmd=$(printf '%s' \
+      "./gradlew :webapp:cas-server-webapp-${project}:build \
+      -DskipNestedConfigMetadataGen=true -x check -x test -x javadoc ${BUILD_CACHE} --configure-on-demand --parallel \
+      ${BUILD_SCRIPT} ${DAEMON} -DcasModules="${dependencies}" --no-watch-fs --max-workers=${MAX_WORKERS} ${BUILDFLAGS}")
+  echo $buildcmd
+  $buildcmd > build.log 2>&1 &
+  pid=$!
+  printcyan "Waiting for build to finish. Process id is ${pid} - Waiting for ${targetArtifact}"
+  counter=0
+  until [[ -f ${targetArtifact} ]]; do
+     let counter++
+     if [[ $counter -gt 60 ]]; then
+        printred "\nBuild taking longer then 15 minutes, aborting."
+        printred "Build log"
+        cat build.log
+        printred "Build thread dump"
+        jstack $pid || true
+        exit 3
+     fi
+     echo -n '.'
+     sleep 15
+  done
+  wait $pid
+  if [ $? -ne 0 ]; then
     printred "\nFailed to build CAS web application. Examine the build output."
-    exit 1
+    cat build.log
+    exit 2
+  else
+    printgreen "\nBackground build successful. Build output was:"
+    cat build.log
```

Comment:
Checking shortly after start of build to see if java is running seems to make jobs on windows more reliable, otherwise in a several random builds, the build doesn't finish but the process id is not running. Build log is empty and pid is gone.